### PR TITLE
add esp32 gfx impl

### PR DESCRIPTION
**Pull request by q349wang**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -203,3 +212,5 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
     target_link_libraries(sungazer PRIVATE SDL2::SDL2)
   endif()
 endif()
+
```
Comment (Victor-C-Zhang) [inline CMakeLists.txt:215]: i hate this, but I guess if you're gonna change it right away it's fine for now

File: espgazer/main.cpp [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  }
+
+extern "C" void app_main(void) {
+  printf("Hello world!\n");
+
+  /* Print chip information */
+  esp_chip_info_t chip_info;
+  uint32_t flash_size;
+  esp_chip_info(&chip_info);
+  printf("This is %s chip with %d CPU core(s), %s%s%s%s, ", CONFIG_IDF_TARGET, chip_info.cores,
```
Comment (Victor-C-Zhang) [inline espgazer/main.cpp]: does this go to a different place than ESP_LOG?

File: espgazer/main.cpp [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+#include "esp_flash.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "sdkconfig.h"
+
+using namespace antioch::gfx;
+#define CHECK_ERROR(expr)         \
+  if (expr != Result::eSuccess) { \
+    printf("%s failed\n", #expr); \
+  }
```
Comment (Victor-C-Zhang) [inline espgazer/main.cpp:26]: ESP_ERROR_CHECK aborts the program. should this too?

File: gfx/CMakeLists.txt [hunk 4]
```diff
@@ -1,12 +1,23 @@
 
 FILE(GLOB BASE_SRCS src/base/*.cpp)
-FILE(GLOB IMPL_SDL2_SRCS src/sdl2/*.cpp)
 
+if(ESP_PLATFORM)
+
+FILE(GLOB ESP32_SRCS src/esp32/*.cpp)
+
+idf_component_register(SRCS ${ESP32_SRCS} ${BASE_SRCS}
```
Comment (Victor-C-Zhang) [inline gfx/CMakeLists.txt:8]: Didn't know the IDF was working with Putin and Xi too

File: gfx/CMakeLists.txt [hunk 5]
```diff
@@ -1,12 +1,23 @@
...
+                       INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/src/base/include ${CMAKE_CURRENT_SOURCE_DIR}/src/esp32/include ${CMAKE_CURRENT_SOURCE_DIR}/include
+                       REQUIRES esp32-hub75 heap)
+
+# Simulator backend
+else()
 
 add_compile_options(
   -fno-exceptions
```
Comment (Victor-C-Zhang) [inline gfx/CMakeLists.txt:16]: does this need to go into the esp build config?

File: sdkconfig [hunk 6]
```diff
@@ -0,0 +1,1936 @@
+#
```
Comment (Victor-C-Zhang) [inline sdkconfig:1]: does this need to be committed?

File: CMakeLists.txt (hunk 1 above)
Comment (q349wang) [reply CMakeLists.txt:215]: its gonna change a bit but i don't think you're gonna like it

File: espgazer/main.cpp (hunk 2 above)
Comment (q349wang) [reply espgazer/main.cpp]: idk what ESP_LOG is

File: gfx/CMakeLists.txt (hunk 4 above)
Comment (q349wang) [reply gfx/CMakeLists.txt:8]: kek

File: espgazer/main.cpp (hunk 3 above)
Comment (q349wang) [reply espgazer/main.cpp:26]: yea sure

File: sdkconfig (hunk 6 above)
Comment (q349wang) [reply sdkconfig:1]: I would say yea, there are some non-default config stuff in here such as the board target we have.

File: espgazer/main.cpp (hunk 2 above)
Comment (Victor-C-Zhang) [reply espgazer/main.cpp]: like alog in android. use that instead of printf bc that can get compiled away in prod

File: gfx/CMakeLists.txt (hunk 5 above)
Comment (q349wang) [reply gfx/CMakeLists.txt:16]: No it does it automatically

File: espgazer/main.cpp (hunk 2 above)
Comment (q349wang) [reply espgazer/main.cpp]: ok

**Review by Victor-C-Zhang — APPROVED** [review]

**Pull request merged**